test_programs/compile_success_empty/comptime_type/src/main.nr

```diff
@@ -66,5 +70,17 @@ fn main() {
         let yes = true;
         let bool_type = type_of(yes);
         assert(bool_type.is_bool());
+
+        // Check Type::as_struct
+        assert(u8_type.as_struct().is_none());
+
+        let foo = Foo { x: 0 };
+        let foo_type = type_of(foo);
+        let (struct_definition, generics) = foo_type.as_struct().unwrap();
+        let fields = struct_definition.fields();
+        assert_eq(fields.len(), 1);
```

Comment:
If I check `fields[0].1`, which is the type of the field, I get it's `T`. I wanted to assert that's it's the field type. Is the current behavior okay? I didn't know if `as_struct()` would return the uninstantiated struct type, or the instantiated one (it currently returns the uninstantiated one).

Reply:
The only way to get the instantiated one is to use the generics to re-instantiate it. Ie. wrap it in a `Type::Struct` again.
We could add a method `get_struct_fields(self) -> Option<[(Quoted, Type)]>` on `Type` to get the instantiated struct field's.